### PR TITLE
Debot interfaces update

### PR DESCRIPTION
**Pull request by mikhailUshakoff**
Add three unstable functions to debot sdk interface. they are proxy to sdk functions:
nacl_box
nacl_box_open
nacl_box_keypair_from_secret_key

Add debot system interface for hex encoding and decoding.

Update tests for internal interfaces.

### CODE REVIEW

**Review by AlexeyVavilin — APPROVED** [review]

File: ton_client/src/debot/hex_interface.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+impl HexInterface {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    fn encode(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        //let data_to_encode = hex::decode(&get_arg(args, "data")?).map_err(|e| format!("{}", e))?;
+		//let encoded = base64::encode(&data_to_encode);
```
Comment (Keshoid) [inline ton_client/src/debot/hex_interface.rs:59]: Please, remove unused code.

File: ton_client/src/debot/hex_interface.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            answer_id,
+            json!({ "hexstr": hex::encode(encoded.as_bytes()) }),
+        ))
+    }
+
+    fn decode(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let str_to_decode = get_string_arg(args, "hexstr")?;
+        let decoded =
+            hex::decode(&str_to_decode).map_err(|e| format!("invalid base64: {}", e))?;
```
Comment (Keshoid) [inline ton_client/src/debot/hex_interface.rs:71]: Fix error message: `invalid base64`. It is not a base64, right?

File: ton_client/src/debot/sdk_interface.rs [hunk 3]
```diff
@@ -436,7 +475,63 @@ impl SdkInterface {
             answer_id,
             json!({ "substr": hex::encode(sub_str.as_bytes()) })
         ))
+    }
 
+    fn nacl_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let decrypted = base64::encode(&hex::decode(&get_arg(args, "decrypted")?).unwrap());
```
Comment (Keshoid) [inline ton_client/src/debot/sdk_interface.rs:482]: Use `map_err()?` instead of `unwrap()`, ok?

File: ton_client/src/debot/sdk_interface.rs [hunk 4]
```diff
@@ -436,7 +475,63 @@ impl SdkInterface {
...
+        let nonce = get_arg(&args, "nonce")?;
+        let public = decode_abi_bigint(&get_arg(&args, "publicKey")?).map_err(|e| e.to_string())?;
+        let secret = decode_abi_bigint(&get_arg(&args, "secretKey")?).map_err(|e| e.to_string())?;
+        let result = nacl_box(
+            self.ton.clone(),
+            ParamsOfNaclBox {                
+                decrypted:decrypted,
+                nonce:nonce,
+                their_public: format!("{:064x}", public),
```
Comment (Keshoid) [inline ton_client/src/debot/sdk_interface.rs:491]: You can write
```
ParamsOfNaclBox {
   decrypted,
   nonce,
 ....
 }
```
If name of var and field are the same.

File: ton_client/src/debot/sdk_interface.rs [hunk 5]
```diff
@@ -436,7 +475,63 @@ impl SdkInterface {
...
+            self.ton.clone(),
+            ParamsOfNaclBoxOpen {                
+                encrypted:encrypted,
+                nonce:nonce,
+                their_public: format!("{:064x}", public),
+                secret: format!("{:064x}", secret),
+            },
+        )
+        .map_err(|e| format!("{}", e))?;
+        Ok((answer_id, json!({ "decrypted": hex::encode(&base64::decode(&result.decrypted).unwrap()) })))
```
Comment (Keshoid) [inline ton_client/src/debot/sdk_interface.rs:515]: Please, handle `unwrap()`.

**Review by Keshoid — CHANGES_REQUESTED** [review]

**Review by Keshoid — APPROVED** [review]
LGTM

**Pull request merged**